TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1580,3 +1757,82 @@ def parse_layer_as_float(shape, properties, fid, zoom):
             properties['layer'] = layer_float
 
     return shape, properties, fid
+
+
+def drop_features_where(
+        feature_layers, zoom, source_layer=None, start_zoom=0,
+        property_name=None, drop_property=True,
+        geom_types=None):
+    """
+    Drops some features entirely when they have a property
+    named `property_name` and its value is true. Also can
+    drop the property if `drop_property` is truthy. If
+    `geom_types` is present and not None, then only types in
+    that list are considered for dropping.
+
+    This is useful for dropping features which we want to use
+    earlier in the pipeline (e.g: to generate points), but
+    that we don't want to appear in the final output.
+    """
+
+    assert source_layer, 'drop_features_where: missing source layer'
+    assert property_name, 'drop_features_where: missing property name'
+
+    if zoom < start_zoom:
+        return None
+
+    layer = _find_layer(feature_layers, source_layer)
+    if layer is None:
+        return None
+
+    new_features = []
+    for feature in layer['features']:
+        shape, properties, fid = feature
+
+        matches_geom_type = \
+            geom_types is None or \
+            shape.geom_type in geom_types
+
+        val = None
```

Comment:
I'm never quite sure of the scoping rules - if `val` isn't declared in the outer scope, is it definitely going to be defined by either branch?

I guess this is because I'm used to `if` statements introducing scope, e.g:

```C
if (something) {
  int val = 1;
} else {
  int val = 2;
}
val; // <-- undefined here, would be a compilation error
```